From 9f472ea1501f8deb0109ea61feda6b8a5eec4f6a Mon Sep 17 00:00:00 2001
From: Peter Wu <peter@lekensteyn.nl>
Date: Tue, 30 Oct 2018 14:31:39 +0100
Subject: [PATCH] CMake: fix broken oss-fuzz build due to the -pie option

check_function_exists (as used by FindZLIB.cmake) seems to fail with the
-pie option as well, do not try to enable it when building for oss-fuzz.

Change-Id: I7d7e0fce1972483a14ac0a91a9f144f22c5ae8a0
Fixes: v2.9.0rc0-2349-g895ad30b5a ("CMake: Fix -pie linker test")
Reviewed-on: https://code.wireshark.org/review/30431
Petri-Dish: Peter Wu <peter@lekensteyn.nl>
Tested-by: Petri Dish Buildbot
Reviewed-by: Peter Wu <peter@lekensteyn.nl>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 374b3f4d7e4..44bce81e383 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -840,7 +840,7 @@ endif()
 
 include(CheckCLinkerFlag)
 
-if(NOT CMAKE_C_COMPILER_ID MATCHES "MSVC")
+if(NOT CMAKE_C_COMPILER_ID MATCHES "MSVC" AND NOT OSS_FUZZ)
 	#
 	# The -pie linker option produces a position-independent executable.
 	# Some Linux distributions have this enabled by default in the compiler,
